### DISCUSSION TITLE
Magick.Net API Documentation

### DISCUSSION THREAD

**Discussion by daryl-williams** [dlemstra/Magick.NET#1420] [Help]
Hello Everyone,

I am new to Magick.Net, and I am trying to learn how to use it but I am unable to find any documentation for methods and functions provided by Magick. Right now I am trying to crop an image and I have searched online using Google, DuckDuckGo, and Bing but I have been unable to find any documentation for the Crop method signature. I have seen a few examples where it appears to take two arguments but nowhere I have I found what these arguments should be or if there is more than one way to call the method. I don't have a lot of experience with imaging but I would have thought that a crop function would take four arguments, i.e. top-left-x, top-left-y, bottom-right-x, bottom-right-y.

Any help would be much appreciated, even if there is no documentation it would be nice to know that. And if there is where is it, or is one supposed dig through the source code to find answers?

Regards,

Daryl

**Comment by dlemstra** [comment]
The VisualStudio intelisense should provide you with some information. The overload that you are looking for is the one that uses a `IMagickGeometry`. And the `MagickGeometry` has a constructor where you can specify the `x,y,width,height`. Where `x` and `y` are top-left and the bottom right position is based on the specified `width` and `height`.

**Comment by daryl-williams** [comment]
Hi @dlemstra, Thank you for your response, I appreciate the information. I guess I should have mentioned that I'm kinda of a  dinosaur and I work (mostly) from the command line so I do miss out on whatever goodies are available through VisualStudio and other IDEs. Thanks again.

D
